extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/binder/kafka/KafkaEventObserver.java

```diff
@@ -0,0 +1,71 @@
+package io.quarkus.micrometer.runtime.binder.kafka;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.producer.Producer;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.binder.kafka.KafkaClientMetrics;
+
+@ApplicationScoped
+public class KafkaEventObserver {
+    MeterRegistry registry = Metrics.globalRegistry;
+    ConcurrentHashMap<Object, KafkaClientMetrics> clientMetrics = new ConcurrentHashMap<>();
+
+    /**
+     * Manage bind/close of KafkaClientMetrics for the specified Consumer.
+     * If the consumer has not been seen before, it will be bound to the
+     * Micrometer registry and instrumented using a Kafka MeterBinder.
+     * If the consumer has been seen before, the MeterBinder will be closed.
+     * 
+     * @param consumer Observed Kafka Consumer
+     */
+    public void consumerCreated(@Observes Consumer<?, ?> consumer) {
+        KafkaClientMetrics prev = clientMetrics.remove(consumer);
+        if (prev == null) {
+            clientMetrics.computeIfAbsent(consumer, x -> {
```

Comment:
Why do you remove it and then computeIfAbsent? Do you expect concurrent access? If I'm not sure it works.

Reply:
it might be safer to just synchronized the whole method.

Reply:
I can change that. I like computeIfAbsent as an atomic operation (personal bias), but I can see that it is confusing.